server/apps/archived/archived.py

```diff
@@ -164,36 +164,44 @@ def on_delete(self, doc):
         doc[config.ID_FIELD] = id_field
 
     def delete(self, lookup):
+        if app.testing and len(lookup) == 0:
+            super().delete(lookup)
+            return
+
+    def on_deleted(self, doc):
+        pass
+
+    def update(self, id, updates, original):
```

Comment:
I would create another endpoint (e.g. /archived/<id>/kill) with CREATE method to mark the archived item for kill. Use patch to apply the template and finally delete to complete the kill process. Using DELETE for checking is not very REST compliant. There's also an edge case, where the archived item could change between the moment when you checked and the moment when you apply the template - and the validations for DELETE could fail.